config/src/main/java/com/typesafe/config/ConfigRenderOptions.java

```diff
@@ -174,9 +204,53 @@ public String toString() {
             sb.append("formatted,");
         if (json)
             sb.append("json,");
+        if (comparator != null)
+            sb.append(comparator.getClass().getSimpleName());
         if (sb.charAt(sb.length() - 1) == ',')
             sb.setLength(sb.length() - 1);
         sb.append(")");
         return sb.toString();
     }
+
+    // this is only Serializable to chill out a findbugs warning
+    static final private class DefaultComparator implements java.util.Comparator<String>, Serializable {
+        private static final long serialVersionUID = 1L;
+
+        private static boolean isAllDigits(String s) {
+            int length = s.length();
+
+            // empty string doesn't count as a number
+            if (length == 0)
+                return false;
+
+            for (int i = 0; i < length; ++i) {
+                char c = s.charAt(i);
+
+                if (Character.isDigit(c))
+                    continue;
+                else
+                    return false;
+            }
+            return true;
+        }
+
+        // This is supposed to sort numbers before strings,
+        // and sort the numbers numerically. The point is
+        // to make objects which are really list-like
+        // (numeric indices) appear in order.
+        @Override
+        public int compare(String a, String b) {
+            boolean aDigits = isAllDigits(a);
+            boolean bDigits = isAllDigits(b);
+            if (aDigits && bDigits) {
+                return Integer.compare(Integer.parseInt(a), Integer.parseInt(b));
```

Comment:
I know you're copying this from our existing code, but I think this will throw if the digits make an integer greater than maxint. We probably ought to fix that... (in its own commit and maybe own PR...)

Reply:
You're totally right. I've changed that to BigInteger to avoid an exception.

Reply:
I believe I am hitting the exact same issue described here while rendering some config. Is this change part of the latest 1.3.3 release? I don't see it on the milestone page yet.

````Exception in thread "main" java.lang.NumberFormatException: For input string: "10000951726339"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:583)
	at java.lang.Integer.parseInt(Integer.java:615)
	at com.typesafe.config.impl.SimpleConfigObject$RenderComparator.compare(SimpleConfigObject.java:452)```

Reply:
I see it is a candidate for the next release. Please ignore the previous comment.

https://github.com/lightbend/config/issues/541